### ISSUE TITLE
Found ';' while parsing initializer list! for member function pointer 

### ISSUE THREAD

**Issue by swiesheier** [doxygen/doxygen#10369]
I use doxygen 1.9.5 and get

`warning: Found ';' while parsing initializer list! (doxygen could be confused by a macro call without semicolon)`

for the docu of a member function pointer in c++

```
    name::ClassB<2, 3, NumberType>(
              ClassA<dim, NumberType>::*myPtr)(const name::ClassB<2,3,NumberType> &foo) const;
```

If I remove the template brackets for the return type,

```
    name::ClassB(
                  ClassA<dim, NumberType>::*myPtr)(const name::ClassB<2,3,NumberType> &foo) const;
```

doxygen treats it as a function and reports
```

    warning: parameters of member myPtr are not documented
    warning: return type of member myPtr is not documented
```

Is there a preprocessor trick to work around this issue?

UPDATE:

I attached a minimal example demonstrating the problem!

[mwe.cc.tar.gz](https://github.com/doxygen/doxygen/files/13042909/mwe.cc.tar.gz)

**Comment by albert-github** [comment]
Note: current doxygen version is 1.9.8


- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by swiesheier** [comment]
@albert-github I added an example that shows the problem.

Unfortunately, I can only use 1.9.5 at the moment.

**Comment by albert-github** [comment]
I see the warning with the versions 1.9/5, 1.9.8 and the current master (1.10.0 (81b651867b4f516a7a266b2acf7d814ab218a834)) also these versions have no relevant differences regarding this problem (as far as I can see quickly)

**Comment by albert-github** [comment]
Interesting observation, might have to do with the handling of the `<` and `>` of the `test::Foo<3,NumberType>` of the line `test::Foo<3,NumberType>(Foo<dim, NumberType>::*funcPtr)(`

**Comment by swiesheier** [comment]
I do not know. 

Do you have a clue for a solution?

**Comment by albert-github** [comment]
It is a difficult problem especially due to all the new "improvements" in the different standards that complicate things here (and also for compiler builders etc.) .

**Comment by doxygen** [comment]
The easiest workaround (that also helps to make the code more readable is to define a type), e.g.

```c++
            /** Name for this type */
            using ThisType = Foo<dim,NumberType>;
            /** 3D Type */
            using Type3D = Foo<3,NumberType>;

            /**
            * Pointer to member func 1
            */
            Typ3D(ThisType::*funcPtr)(const Type3D &arg1) const;
```

**Comment by doxygen** [comment]
@swiesheier Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official version is released.

**Comment by swiesheier** [comment]
> @swiesheier Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official version is released.

A typedef fixes the issue in this minimal example, however, I do not want to define a type in my real code.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
